src/components/app/app.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { fireEvent, render, screen, within } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { App } from './app'
+import React from 'react'
+import * as deck from '../../services/deck'
+import { CardSuit, CardValue } from '../../interfaces/card.interface'
+
+test('Renders app', () => {
+    render(<App />)
+
+    expect(screen.getByText('DEALER')).toBeInTheDocument()
+    expect(screen.getByText('PLAYER')).toBeInTheDocument()
+})
+
+test('Dealer is dealt a seven and ten', () => {
+    mockDealHand(CardValue.SEVEN, CardValue.TEN)
+    mockDealHand(CardValue.SEVEN, CardValue.TEN)
+
+    render(<App />)
+
+    fireEvent.click(screen.getByText('STAND'))
+
+    expect(
+        screen.getByText('DEALER').nextSibling.firstChild.nextSibling
```

Comment:
we do not want to be relying on nested calls to DOM structure like `nextSibling`, `firstChild`, etc. It is not very readable but also very fickle and likely to break when structure of layout changes with upcoming features.

We could look into enhancing components like `CardTotalDisplay` to take in a `datatest-id: string` of sorts that would be attribute we add onto elements but let's not make that big move for now.

Instead let's continue to look to use `screen.getByText`.

- update CardTotalDisplay component to render the contents of total inside the _same_ `<span>` tag as the text `Total:`
 ```
  // update to look like this now
                <div className={`${styles.standard} ${styles.leftSpacing}`}>
                    <span className={styles.total}>Total: 
                        {cardTotal.isSoft && cardTotal.total !== 21
                            ? 'Soft '
                            : null}{' '}
                        {cardTotal.total}
                    </span>
                </div>
 ```
- update this test so that the dealer and player do not have same value. assert on both totals being visible
  ```
    test('Player and dealer totals are displayed after finishing game', () => {
        mockDealHand(CardValue.SEVEN, CardValue.TEN)
        mockDealHand(CardValue.SEVEN, CardValue.SEVEN)

        render(<App />)

        fireEvent.click(screen.getByText('STAND'))

        expect(screen.getByText('Total: 14')).toBeVisible();
        expect(screen.getByText('Total: 17')).toBeVisible();
    })
  ```

if you ever have a test in which you _need_ to have both players with same total, then use `screen.getAllByText()` to get back array of results and can assert on individual entries.